File: Level 3/기지국 설치/solution.js
function solution(n, stations, w) {
  let answer = 0;
  let location = 1;
  let index = 0;

  while (location <= n) {
    if (index < stations.length && location >= stations[index] - w) {
      location = stations[index] + w + 1;
      index++;
    } else {
      location += w * 2 + 1;
      answer++;
    }
  }

  return answer;
}

File: Level 3/기지국 설치/solution.py
def solution(n, stations, w):
    answer = 0
    location = 1
    idx = 0

    while location <= n:
        if idx < len(stations) and location >= stations[idx] - w:
            location = stations[idx] + w + 1
            idx += 1
        else:
            location += w * 2 + 1
            answer += 1

    return answer
